docs/storage.md

```diff
@@ -110,6 +111,8 @@ Please refer to the documentation of [the Transport type](https://golang.org/pkg
 
 `part_size` is specified in bytes and refers to the minimum file size used for multipart uploads, as some custom S3 implementations may have different requirements. A value of `0` means to use a default 128 MiB size.
 
+Set `list_objects_v1: true` for S3 compatible APIs that don't support ListObjectsV2 (e.g. some versions of Ceph).
```

Comment:
I would vote for `list_objects_version: "v1"` with default being `"v2"`. It might be much more flexible, WDYT? (:

Reply:
So @bwplotka should I make the switch? I'll defer to your opinion.

My only preference for this PR is that it gets merged into main sooner rather than later so I can deploy a real upstream Thanos (and not my own binary :))

Reply:
Yes please! Let's switch. It's a good practice overall for the future (what if we change S3 client to different one?) 

That's the last suggestion, I think once it's done we are good to merge :hugs:

Reply:
Question. I am choosing to set it as a string in the `type Config struct`, and leave it as a bool in `type Bucket string` (which contains a minio.Client as a field, so it makes sense to have a minio-specific in that struct).

The function NewBucketWithConfig is a good place to store the logic of parsing `""|"v2"` or `v1` into the appropriate bool value. Should I break the parsing logic into a function so I can unit test it?
```
var useListObjectsV1 bool
if config.ListObjectsVersion == "" || config.ListObjectsVersion == "v2" {
    useListObjectsV1 = false
} else if config.ListObjectsVersion == "v1" {
    useListObjectsV1 = true
} else {
    listObjectsErrMsg := errors.Errorf("Unsupported version %q was provided. Supported values are v1, v2", config.ListObjectsVersion)
    return nil, errors.Wrap(listObjectsErrMsg, "Initialize s3 client list objects version")
}
```

Reply:
It's more clear if I push commit (just did).